In [71]:
df = pd.read_csv('C:/Users/user/Downloads/FoodBalanceSheets_E_Africa_NOFLAG_new.csv',encoding='latin-1')

In [174]:
df.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

**Q11** Answer the following questions based on the African food production dataset provided by the FAO website already provided

What is the total sum of Animal Fat produced in 2014 and 2017 respectively? Hint: Perform a groupby sum aggregation on ‘Item’

In [176]:
df_new=df[['Item','Y2014','Y2017']]
df_new.groupby(['Item']).sum()

Y2014      Y2017
Item                                     
Alcohol, Non-Food      2403.00    2348.00
Alcoholic Beverages  102410.11   95581.06
Animal Products       11935.65   11547.65
Animal fats          209460.54  269617.53
Apples and products    9499.23   10198.90
...                        ...        ...
Vegetables, Other    155038.96  157752.59
Vegetal Products     107145.19  107655.20
Wheat and products   232670.13  240047.62
Wine                   4497.36    4178.02
Yams                 200396.96  229174.59

[119 rows x 2 columns]

**Q12** What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [191]:
print('Mean:',round(df['Y2015'].mean(),3))
print('Standard Deviation:',round(df['Y2015'].std(),3))

Mean: 135.236
Standard Deviation: 1603.404


**Q13** What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [202]:
total = df['Y2016'].count()

In [203]:
null_counts = df['Y2016'].isnull().sum()
print("Total nulls:",null_counts)

Total nulls: 1535


**Q14** Which year had the highest correlation with ‘Element Code’?  

 >Ans: Its 2014

In [104]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000


Q15
What year has the highest sum of Import Quantity?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question 


> Ans 2017


In [109]:
df.groupby(['Element']).sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

**Q16** What is the total number of the sum of Production in 2014?

Hint- Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question

>Ans 1931287.75

In [111]:
df.groupby(['Element']).sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

**Q17** Which of these elements had the highest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.

>Ans Domestic supply quantity 	

In [120]:
temp = df[['Element','Y2018']]
temp

Element     Y2018
0               Total Population - Both sexes  42228.00
1                    Domestic supply quantity      0.00
2               Food supply (kcal/capita/day)   3322.00
3      Protein supply quantity (g/capita/day)     91.83
4          Fat supply quantity (g/capita/day)     77.28
...                                       ...       ...
60938                                    Food     16.00
60939     Food supply quantity (kg/capita/yr)      1.08
60940           Food supply (kcal/capita/day)      1.00
60941  Protein supply quantity (g/capita/day)      0.04
60942      Fat supply quantity (g/capita/day)      0.01

[60943 rows x 2 columns]

In [124]:
temp.groupby('Element').sum()

Y2018
Element                                           
Domestic supply quantity                2161192.10
Export Quantity                          181594.80
Fat supply quantity (g/capita/day)        10258.69
Feed                                     233489.68
Food                                    1303841.28
Food supply (kcal/capita/day)            455261.00
Food supply quantity (kg/capita/yr)       49056.85
Import Quantity                          287997.09
Losses                                   163902.00
Other uses (non-food)                     91300.97
Processing                               308429.00
Production                              2075072.89
Protein supply quantity (g/capita/day)    11833.56
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Total Population - Both sexes           1140605.00
Tourist consumption                          90.00

**Q18**Which of these elements had the 3rd lowest sum in 2018?

Hint-  Select columns ‘Y2018’ and ‘Element’, Perform a groupby operation on ‘Element’ on the selected dataframe and answer the  question.  
> Protein suppy

In [125]:
new_temp =temp.groupby('Element').sum()

In [208]:
new_temp

Y2018
Element                                           
Domestic supply quantity                2161192.10
Export Quantity                          181594.80
Fat supply quantity (g/capita/day)        10258.69
Feed                                     233489.68
Food                                    1303841.28
Food supply (kcal/capita/day)            455261.00
Food supply quantity (kg/capita/yr)       49056.85
Import Quantity                          287997.09
Losses                                   163902.00
Other uses (non-food)                     91300.97
Processing                               308429.00
Production                              2075072.89
Protein supply quantity (g/capita/day)    11833.56
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Total Population - Both sexes           1140605.00
Tourist consumption                          90.00

**Q19**What is the total Import Quantity in Algeria in 2018?
> Ans 36238.29

In [156]:
df[df['Area']=="Algeria"].groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                      464     311204        614916   
Export Quantity                               392     264011        579278   
Fat supply quantity (g/capita/day)            432     291489         73872   
Feed                                           88      58967        121462   
Food                                          420     282744        539910   
Food supply (kcal/capita/day)                 432     291489         71712   
Food supply quantity (kg/capita/yr)           420     282744         67725   
Import Quantity                               448     301030        628432   
Losses                                        208     139857        266396   
Other uses (non-food)                         148     101269        190698   
Processing                                    140      93537        179585   
Production                                    360     242590        495990   
Protein supply quantity (g/capita/day)        432     291489         72792   
Residuals                                     408     272327        527340   
Seed                                           80      53109        110540   
Stock Variation                               376     251696        476768   
Total Population - Both sexes                   4       2501           511   

                                           Y2014     Y2015     Y2016  \
Element                                                                
Domestic supply quantity                83755.94  85972.86  87157.61   
Export Quantity                          1328.64   1104.62   1340.22   
Fat supply quantity (g/capita/day)        320.19    317.40    309.54   
Feed                                     9656.04  10308.00   9903.00   
Food                                    60582.69  61693.49  63081.43   
Food supply (kcal/capita/day)           13500.00  13512.00  13483.00   
Food supply quantity (kg/capita/yr)      1556.45   1552.81   1555.77   
Import Quantity                         34940.12  37841.91  36736.08   
Losses                                   7231.00   7471.00   7405.00   
Other uses (non-food)                    3315.22   3483.37   3400.17   
Processing                               1776.00   1865.00   1920.00   
Production                              52534.48  53451.57  54554.75   
Protein supply quantity (g/capita/day)    379.56    377.37    378.84   
Residuals                                 552.00    489.00    770.00   
Seed                                      631.00    657.00    672.00   
Stock Variation                          2392.00   4233.00   2789.00   
Total Population - Both sexes           38924.00  39728.00  40551.00   

                                           Y2017     Y2018  
Element                                                     
Domestic supply quantity                86810.42  90023.42  
Export Quantity                          1494.54    811.54  
Fat supply quantity (g/capita/day)        320.74    309.09  
Feed                                     9571.00  10668.00  
Food                                    62785.33  63736.33  
Food supply (kcal/capita/day)           13363.00  13288.00  
Food supply quantity (kg/capita/yr)      1516.94   1509.28  
Import Quantity                         36059.29  36238.29  
Losses                                   7428.00   7708.00  
Other uses (non-food)                    3534.08   3970.08  
Processing                               2045.00   2479.00  
Production                              54869.67  60766.67  
Protein supply quantity (g/capita/day)    371.26    367.28  
Residuals                                 766.00    630.00  
Seed                                      684.00    825.00  
Stock Variation                          2636.00   6172.00  
Total Population - Both sexes           41389.00  42228.00

In [218]:
**Q2** What is the total number of unique countries in the dataset?

df['Area'].nunique()

Object `dataset` not found.


49